test/xpu_api/numerics/numeric.ops/numeric.ops.gcd/xpu_gcd.pass.cpp

```diff
@@ -0,0 +1,150 @@
+
+    {
+        cl::sycl::buffer<bool, 1> buffer1(&res, numOfItems1);
+        deviceQueue.submit([&](cl::sycl::handler& cgh) {
+            auto out = buffer1.get_access<sycl_write>(cgh);
+            cgh.single_task<KernelTest>([=]() {
+                using S1 = oneapi::dpl::make_signed_t<Input1>;
+                using S2 = oneapi::dpl::make_signed_t<Input2>;
+                using U1 = oneapi::dpl::make_unsigned_t<Input1>;
+                using U2 = oneapi::dpl::make_unsigned_t<Input2>;
+                constexpr struct
+                {
+                    int x;
+                    int y;
+                    int expect;
+                } Cases[] = {{0, 0, 0}, {1, 0, 1}, {0, 1, 1},   {1, 1, 1},
+                             {2, 3, 1}, {2, 4, 2}, {36, 17, 1}, {36, 18, 18}};
+                for (auto TC : Cases)
+                {
+                    { // Test with two signed types
+                        using Output = oneapi::dpl::common_type_t<S1, S2>;
+                        out[0] &= test0<S1, S2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(-TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(-TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(-TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(-TC.x, -TC.y, TC.expect);
+                    }
+
+                    { // test with two unsigned types
+                        using Output = oneapi::dpl::common_type_t<U1, U2>;
+                        out[0] &= test0<U1, U2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, U1, Output>(TC.x, TC.y, TC.expect);
+                    }
+                    { // Test with mixed signs
+                        using Output = oneapi::dpl::common_type_t<S1, U2>;
+                        out[0] &= test0<S1, U2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, S1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, S1, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S1, U2, Output>(-TC.x, TC.y, TC.expect);
+                    }
+                    { // Test with mixed signs
+                        using Output = oneapi::dpl::common_type_t<S2, U1>;
+                        out[0] &= test0<S2, U1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U1, S2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U1, S2, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, U1, Output>(-TC.x, TC.y, TC.expect);
+                    }
+                }
+                { //  LWG#2837
```

Comment:
What does this comment mean?

Reply:
This test is ported from libc++ testsuite, I just keep it here. 
Now I have removed this comment.